datastore/elastic_search/query.py

```diff
@@ -312,6 +314,14 @@ def full_text_query(connection, index_name, doc_type, entity_name, sentences, fu
     try:
         response = _run_es_search(connection, msearch=True, **kwargs)
         results = _parse_es_search_results(response.get("responses"))
+    except es_exceptions.NotFoundError as e:
+        raise DataStoreRequestException(f'NotFoundError in datastore query on index: {index_name}',
```

Comment:
Is this one supposed to be an error? I'm asking because I think we query every entity against all messages and there might be a lot of cases wher e it isn't found. If that is not the case then no change required.